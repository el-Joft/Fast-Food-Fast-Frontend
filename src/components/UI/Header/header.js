import React from 'react';
import {Link} from 'react-router-dom';
import Navbar from '../Navbar/Navbar';
import './header.scss';

const Header = () => {
  const logo = () => {
    return <div className='logo'>
      <Link to='/'>
        <img src='/public/images/fast-food-fast.png' alt=""/>
        <div className='logo-text'><h2>Fast Food Fast</h2></div>
      </Link>
    </div>
  } 

  const workingHours = () => {
    return <div className="working-hours">
    <img className="icons-telephone" src="/public/images/icons/telephone.png" alt="" />
    <address className="contact-info">
        <Link to='callto:#' className="phone">+234-80502000</Link><br/>
        <span className="">Mon – Sat: 9:00–18:00</span>
    </address>
</div>
  }

const socialItem = [
  {
    link: '/facebook',
    src: '/public/images/icons/facebook.png',
    alt: 'facebook',
  },
  {
    link: '/twitter',
    src: '/public/images/icons/twitter.png',
    alt: 'twitter',
  },
  {
    link: '/instagram',
    src: '/public/images/icons/instagram.png',
    alt: 'instagram',
  },
  {
    link: '/google',
    src: '/public/images/icons/google.png',
    alt: 'google',
  },

]
const socialMedia = () => socialItem.map((item, i) => 
  <li key={i}>
  <Link to={item.link}>
    <img src={item.src} alt='item.alt'/>
  </Link>
</li>     
)
  return (
    <div>
    <div className='header-1'>
      {logo()}
      {workingHours()}
      <div className="social">
      <ul>
        {socialMedia()}
      </ul>
      </div>
      </div>
      <Navbar />
      </div>
  )
}

export default Header;