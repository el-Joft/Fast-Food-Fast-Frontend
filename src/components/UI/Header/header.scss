$color_1: #42ce9f;
$color_2: #fff;
$font_family_1: Righteous;
$background_color_1: #595351;

/* HEADER */
/* ICONS */
/* SOCIAL */
/* CONTACT */
.header-1 {
	width: 100%;
	background: #fff;
	height: 140px;
	display: flex;
	padding-top: 10px;
	color: $color_1;
}
.logo {
	margin-left: 40px;
	img {
		height: 100px;
		width: 150px;
		float: left;
	}
	a {
		display: flex;
	}
}
.logo-text {
	font-size: 1.7em;
	margin-top: 20px;
	h2 {
		font-family: $font_family_1 !important;
		color: $color_1;
		margin-top: 12px;
	}
}
.social {
	img {
		width: 30px;
		height: 30px;
		color: $color_2;
		background-color: $background_color_1;
		box-shadow: 0 0 0 4px #d9d9d9;
		border-radius: 80px;
	}
	margin-top: 35px;
	ul {
		display: flex;
		justify-content: flex-end;
	}
	li {
		a {
			margin-right: 15px;
			flex: 1;
		}
	}
}
.icons-telephone {
	width: 30px;
	height: 30px;
	color: $color_2;
	background-color: $background_color_1;
	box-shadow: 0 0 0 4px #d9d9d9;
	border-radius: 80px;
	margin-right: 12px;
	margin-top: 5px;
}
.phone {
	color: $color_1;
}
.working-hours {
	display: flex;
	margin-top: 35px;
}
