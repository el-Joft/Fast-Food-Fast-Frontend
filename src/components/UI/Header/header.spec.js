import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from './header';

describe('shallow testing', () => {
  it('should test the component if it renders anything', () => {
    const wrapper = shallow(<Header />);
    const header = wrapper.find('Navbar');
    expect(header.length).toEqual(1);
  });

  it('should test the component contains the classname header-1', () => {
    const wrapper = shallow(<Header />);
    const classname = wrapper.find('.header-1');
    expect(classname.exists()).toBe(true);
  });

  it('testing the header with snapshots', () => {
    const tree = shallow(<Header />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  // testing lifecycle methods using mount rendering
  // here test can affect each other when they are using the same dom
  // it('should matches the snapshot', () => {
  //   const tree = mount(<Header />);
  //   expect(toJson(tree)).toMatchSnapshot();
  //   tree.unmount();
  // });
});
