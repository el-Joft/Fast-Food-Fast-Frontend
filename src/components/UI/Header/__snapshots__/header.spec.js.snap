// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow testing testing the header with snapshots 1`] = `
<div>
  <div
    className="header-1"
  >
    <div
      className="logo"
    >
      <Link
        replace={false}
        to="/"
      >
        <img
          alt=""
          src={Object {}}
        />
        <div
          className="logo-text"
        >
          <h2>
            Fast Food Fast
          </h2>
        </div>
      </Link>
    </div>
    <div
      className="working-hours"
    >
      <img
        alt=""
        className="icons-telephone"
        src={Object {}}
      />
      <address
        className="contact-info"
      >
        <Link
          className="phone"
          replace={false}
          to="callto:#"
        >
          +234-80502000
        </Link>
        <br />
        <span
          className=""
        >
          Mon – Sat: 9:00–18:00
        </span>
      </address>
    </div>
    <div
      className="social"
    >
      <ul>
        <li
          key="0"
        >
          <Link
            replace={false}
            to="/facebook"
          >
            <img
              alt="item.alt"
              src={Object {}}
            />
          </Link>
        </li>
        <li
          key="1"
        >
          <Link
            replace={false}
            to="/twitter"
          >
            <img
              alt="item.alt"
              src={Object {}}
            />
          </Link>
        </li>
        <li
          key="2"
        >
          <Link
            replace={false}
            to="/instagram"
          >
            <img
              alt="item.alt"
              src={Object {}}
            />
          </Link>
        </li>
        <li
          key="3"
        >
          <Link
            replace={false}
            to="/google"
          >
            <img
              alt="item.alt"
              src={
                Object {
                  "google": Object {},
                }
              }
            />
          </Link>
        </li>
      </ul>
    </div>
  </div>
  <Navbar />
</div>
`;
